pkg/controllers/binding/common.go

```diff
@@ -70,20 +68,15 @@ func ensureWork(
 		targetClusters = bindingObj.Spec.Clusters
 	}
 
-	hasScheduledReplica, desireReplicaInfos := getReplicaInfos(targetClusters)
-
 	var jobCompletions []workv1alpha2.TargetCluster
-	var jobHasCompletions = false
+	var err error
 	if workload.GetKind() == util.JobKind {
-		completions, found, err := unstructured.NestedInt64(workload.Object, util.SpecField, util.CompletionsField)
+		jobCompletions, err = divideReplicasByJobCompletions(workload, targetClusters)
 		if err != nil {
 			return err
 		}
-		if found {
-			jobCompletions = util.DivideReplicasByTargetCluster(targetClusters, int32(completions))
-			jobHasCompletions = true
-		}
 	}
+	hasScheduledReplica, desireReplicaInfos := getReplicaInfos(targetClusters)
```

Comment:
Can we not adjust the position of this line of code so that the whole PR diff looks clearer?

Reply:
Variables closer to the calling function should be more clearly I think, wdyt?